### PR TITLE
Updated notebook generation to new format

### PR DESCRIPTION
**Pull request by cstefanache**
This is what I was able to come up with in order to update the notebook generation to new format.
Please see if there is something that you can use from this code

### CODE REVIEW

File: include/rmlab/Notebook.cpp [hunk 1]
```diff
@@ -75,23 +76,31 @@ namespace detail
         // select 3:   0 (brush)
         // select 4:   5 (marker/highlighter: always color 0)
         // what is/was 6? :-)
+        std::cout << "brush type" << "\n";
```
Comment (ax3l) [inline include/rmlab/Notebook.cpp:79]: please remove the debug output again :)

File: include/rmlab/Notebook.cpp [hunk 2]
```diff
@@ -75,23 +76,31 @@ namespace detail
...
         // brush base size: 1.875, 2.0, 2.125
         fstream.read( (char*)&curLine.brush_base_size, sizeof(float) );
+        std::cout << "npoints" << "\n";
         fstream.read( (char*)&curLine.npoints, sizeof(int32_t) );
 
-        for( int n = 0; n < curLine.npoints; ++n )
-        {
-            readPoint( fstream, curLine );
-        }
+        // for( int n = 0; n < curLine.npoints; ++n )
```
Comment (ax3l) [inline include/rmlab/Notebook.cpp:99]: Can you please remove the dead code if its not needed anymore?
At which point do we read the actual point data now?

File: include/rmlab/auxiliary/UUID.hpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#ifndef UUID_H
```
Comment (ax3l) [inline include/rmlab/auxiliary/UUID.hpp:1]: let us just stay with pragma once :)
Please also add the license header :)

File: include/rmlab/auxiliary/UUID.hpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+std::string generate_uuid() {
+  return generate_hex(4) + "-" +generate_hex(2) + "-" +generate_hex(2) + "-" +generate_hex(2) + + "-" +generate_hex(6);
+}
+
+}
+}
+
+#endif
```
Comment (ax3l) [inline include/rmlab/auxiliary/UUID.hpp:44]: please remove

File: include/rmlab/auxiliary/UUID.hpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+#include <iostream>
+using namespace std;
+
+namespace rmlab
+{
+namespace auxiliary
+{
+
+
+unsigned int random_char() {
```
Comment (ax3l) [inline include/rmlab/auxiliary/UUID.hpp:18]: can you please add doxygen strings and move the implementation details to a `.cpp` file?

**Review by ax3l — COMMENTED** [review]
Thank you for the pull-request and sorry for my slow response.

I accidentally merged this instead of posting my review comments. (Removed the merge commit on `develop` now to fix.)
Can you please re-open the PR and address the comments?

**Comment by cstefanache** [comment]
Trying to reopen

**Pull request merged**